browser/ui/sidebar/sidebar_model.cc

```diff
@@ -184,64 +176,30 @@ void SidebarModel::SetActiveIndex(int index, bool load) {
   if (index == active_index_)
     return;
 
-  // Don't load url if it's already loaded. If not, new loading is started
-  // whenever item is activated.
-  // TODO(simonhong): Maybe we should have reload option?
-  if (load && index != -1 && !IsLoadedAt(index))
-    LoadURLAt(GetAllSidebarItems()[index].url, index);
-
   UpdateActiveIndexAndNotify(index);
 }
 
-content::WebContents* SidebarModel::GetWebContentsAt(int index) {
-  // Only webcontents is requested for items that opens in panel.
-  // Opens in new tab doesn't need to get webcontents here.
-  DCHECK(GetAllSidebarItems()[index].open_in_panel);
-
-  return data_[index]->GetWebContents();
-}
-
-bool SidebarModel::IsSidebarWebContents(
-    const content::WebContents* web_contents) const {
-  for (const auto& data : data_) {
-    if (data->web_contents() && data->web_contents() == web_contents)
-      return true;
-  }
-
-  return false;
-}
-
-const std::vector<SidebarItem> SidebarModel::GetAllSidebarItems() const {
+const std::vector<SidebarItem>& SidebarModel::GetAllSidebarItems() const {
   return GetSidebarService(profile_)->items();
 }
 
-bool SidebarModel::IsLoadedAt(int index) const {
-  DCHECK(GetAllSidebarItems()[index].open_in_panel);
-
-  return data_[index]->IsLoaded();
-}
-
 bool SidebarModel::IsSidebarHasAllBuiltiInItems() const {
   return GetSidebarService(profile_)->GetHiddenDefaultSidebarItems().empty();
 }
 
 int SidebarModel::GetIndexOf(const SidebarItem& item) const {
   const auto items = GetAllSidebarItems();
   const auto iter =
-      std::find_if(items.begin(), items.end(),
-                   [item](const auto& i) { return item.url == i.url; });
+      base::ranges::find_if(items.begin(), items.end(), [&item](const auto& i) {
```

Comment:
When you use `ranges`, I think you can omit begin(), end() of container.
```suggestion
      base::ranges::find_if(items, [&item](const auto& i) {
```